lnwallet/channel.go

```diff
@@ -3372,14 +3385,32 @@ func (lc *LightningChannel) validateCommitmentSanity(theirLogCounter,
 	ourInitialBalance := commitChain.tip().ourBalance
 	theirInitialBalance := commitChain.tip().theirBalance
 
-	ourBalance, theirBalance, commitWeight, filteredView, err := lc.computeView(
+	ourBalance, theirBalance, commitWeight, feeSiphon, filteredView, err := lc.computeView(
 		view, remoteChain, false,
 	)
 	if err != nil {
 		return err
 	}
 	feePerKw := filteredView.feePerKw
 
+	// For the anchor channelt type, we want to disallow states where the
+	// potential fees siphoned from HTLC timeout transactions are greater
+	// than what is owed to the node in any future states. Since we know
+	// that they must always keep the reserve available for punishment, we
+	// disallow any state where this wouldn't be the case.
+	reserve := lc.channelState.LocalChanCfg.ChanReserve
+	if remoteChain {
+		reserve = lc.channelState.RemoteChanCfg.ChanReserve
+	}
+
+	// TODO(halseth): this limits the number of HTLCs that can be used
+	// together with small channel reserves. Should move to a channel type
+	// where the committed fees for HTLC transactions are zero (only BYOF)
+	// or zero for both commitment+HTLCs (needs package relay).
+	if !lc.skipFeeSiphonCheck && feeSiphon > reserve {
```

Comment:
This'll also cause "silent failures" from the PoV of receivers with this clamp code connected to early adopter nodes that may already be running the existing anchor logic in the wild. However, I guess we can assume that this pairing will be rare in practice, since this upcoming release will be the first one that "enables" anchors network wide from the PoV of `lnd`.